### PR TITLE
Amp Start Configurator Gulp Tasks

### PR DESCRIPTION
**Pull request by torch2424**
This PR is really on **118 additions and 15 deletions**, the `testsTemplates/article.html` can be ignored, and was approved by @camelburrito for testing.

This finishes up the build system for the amp start configurator. This creates all of the gulp tasks requried to build and serve the configurator. And I made it part of the standard build process. Also made gulp tasks for testing, and things of that sort.

# Example

**This example is just me serving the web application itself. On `npm run serve` the web app is watched, and will rebuild prod builds of the configurator. Which is what we want when developing both amp start and the configurator**

`npm run configurator` or `gulp configurator:serve`

configuratorampstart

### CODE REVIEW

File: tasks/configurator.js [hunk 1]
```diff
@@ -13,15 +15,96 @@
...
+const webpackProdConfig = require('../webpack.config.js')({
+  prod: true
+});
+const webpackDevConfig = require('../webpack.config.js')({
+  dev: true
+});
+const webpackTestConfig = require('../webpack.config.js')({
+  test: true
+});
+const WebpackDevServer = require("webpack-dev-server");
```
Comment (erwinmombay) [inline tasks/configurator.js]: use single quote here and anywhere else possible

File: tasks/configurator.js [hunk 2]
```diff
@@ -13,15 +15,96 @@
...
+    }
+
+    // Output the webpack stats
+    util.log("[webpack]", stats.toString({}));
+  });
+}
+
+function runWebpackDevServer(webpackConfig) {
+  if(webpackConfig.entry && Array.isArray(webpackConfig.entry)) {
+    webpackConfig.entry.unshift("webpack-dev-server/client?http://localhost:8080");
```
Comment (erwinmombay) [inline tasks/configurator.js]: lets make the port configurable with the default as 8080

File: tasks/configurator.js [hunk 3]
```diff
@@ -13,15 +15,96 @@
...
+
+function runKarma(options, callback) {
+
+  // Parse our karma config
+  karmaServerConfig = {
+    configFile: `${__dirname}/../karma.conf.js`
+  };
+
+  // Pass in our options
+  Object.keys(options).forEach(function(key) {
```
Comment (erwinmombay) [inline tasks/configurator.js]: We can use Object.assign here i think to make this shorter (depending on support)

Comment (torch2424) [inline tasks/configurator.js]: Hahahahaha! You are right, that is what I get for copying code from the webpack docs. Thanks!

File: tasks/configurator.js [hunk 4]
```diff
@@ -13,15 +15,96 @@
...
 const postcss = require('gulp-postcss');
 const replace = require('gulp-replace');
 const csstree = require('css-tree');
 const rename = require('gulp-rename');
 const through = require('through2');
+const webpack = require("webpack");
+const webpackProdConfig = require('../webpack.config.js')({
+  prod: true
+});
+const webpackDevConfig = require('../webpack.config.js')({
```
Comment (erwinmombay) [inline tasks/configurator.js]: love this! +1

File: tasks/configurator.js (hunk 1 above)
Comment (torch2424) [inline tasks/configurator.js]: Done

File: tasks/configurator.js (hunk 2 above)
Comment (torch2424) [inline tasks/configurator.js]: Done

File: tasks/configurator.js (hunk 3 above)
Comment (torch2424) [inline tasks/configurator.js]: Done

File: tasks/configurator.js (hunk 4 above)
Comment (torch2424) [inline tasks/configurator.js]: Thanks!

**Pull request merged**